### PR TITLE
Support the EU Region

### PR DESCRIPTION
**Pull request by cea2aj**
Add support to the Search, Chat, and Pages analytics reporters for the EU

As part of this change, also add support for Sandbox to Search in order to make it consistent with Chat's support of Sandbox.

The yext-pixel.com domains have been updated to the new domains, and all of the code which constructs the URL endpoints is now consolidated in the provideEndpoints.ts file.

J=BACK-2262
TEST=auto, manual

Update existing tests and add new tests which confirm that the domains are correct when Search is configured for the EU or Sandbox.

Set the region and env in the test site for search, pages, and chat and confirm the requests are fired properly and the network responses are as expected. Check the EU db table for the search events and confirm they are present.

### CODE REVIEW

**Review by EmilyZhang777 — COMMENTED** [review]
Overall lgtm! Just some minor comments

File: src/utils/endpointProviders.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+  return getDomain(CONVERSION_TRACKING_DOMAINS, region, env);
+}
+
+function getEventDomain(region?: Region, env?: Environment): string {
+  return getDomain(EVENT_DOMAINS, region, env);
+}
+
+function getDomain(
+  domainMap: DomainMap,
+  region?: Region,
```
[the remark this answers is not in the document]
Comment (cea2aj) [reply src/utils/endpointProviders.ts]: good call! updated

File: src/infra/SearchAnalyticsReporter.ts [hunk 2]
```diff
@@ -11,9 +10,13 @@ const DEFAULT_DOMAIN = 'https://answers.yext-pixel.com';
 export class SearchAnalyticsReporter implements SearchAnalyticsService {
   private _visitor: Visitor | undefined;
   private _debug: boolean|undefined;
+  private readonly _endpoint: string;
   constructor(private config: SearchAnalyticsConfig, private httpRequesterService: HttpRequesterService) {
     this.setVisitor(config.visitor);
     this._debug = config.debug;
+    this._endpoint = this.config.domain
```
[the remark this answers is not in the document]
Comment (cea2aj) [reply src/infra/SearchAnalyticsReporter.ts]: updated! that cleaned things up

File: src/infra/PagesAnalyticsReporter.ts [hunk 3]
```diff
@@ -154,23 +155,14 @@ export class PagesAnalyticsReporter implements PagesAnalyticsService{
...
-  }
-
   /** {@inheritDoc PagesAnalyticsService.setDebugEnabled} */
   async track(event: PagesAnalyticsEvent, conversionInfo?: ConversionDetails): Promise<void> {
     /** TODO: need to evaluate that the event name is valid, I think there are restrictions in the characters
       * that are accepted
       */
-    const url = new URL(this.endpoint());
+    const endpoint
+      = getPagesEndpoint(this._region, 'PRODUCTION', this._conversionTrackingEnabled);
```
[the remark this answers is not in the document]
Comment (cea2aj) [reply src/infra/PagesAnalyticsReporter.ts]: refactored to not require it!

**Review by benmcginnis — APPROVED** [review]

File: src/models/config/ChatAnalyticsConfig.ts [hunk 4]
```diff
@@ -6,8 +9,8 @@
 export interface ChatAnalyticsConfig {
   /** The API key for accessing the Analytics Events API. */
   apiKey: string; // TODO: Add optional "token" for JWT auth
-  /** The environment to send the requests to. */
-  env?: 'PRODUCTION' | 'SANDBOX';
-  /** The physical region of the Yext account */
-  region?: 'US' | 'EU';
+  /** {@inheritDoc SearchAnalyticsConfig.env} */
+  env?: Environment;
```
Comment (yen-tt) [inline src/models/config/ChatAnalyticsConfig.ts:13]: can we do EnumOrLiteral for env and region?

Comment (cea2aj) [reply src/models/config/ChatAnalyticsConfig.ts:13]: updated!

**Review by yen-tt — APPROVED** [review]

**Pull request merged**